### PR TITLE
Revert "Revert "Use the `until` framework for resolving results""

### PR DESCRIPTION
**Pull request by th0114nd**
Reverts blockapps/blockapps-rest#131

Includes the fixes that 118 was initially lacking

### CODE REVIEW

File: lib/rest_5.js [hunk 1]
```diff
@@ -485,19 +485,13 @@ function* resolveResult(result, chainId, node) {
...
-    while (count < 60 && res.filter(r => {return r.status === constants.PENDING}).length !== 0) {
-        res = yield getBlocResults(res.map(r => {return r.hash}), chainId, false, node);
-	yield promiseTimeout(1000);
-	count++;
-    }
-
-    if(count >= 60) {
-        throw new HttpError400('Transaction did not resolve');
+    const pred = res => !res.some(r => r.status === constants.PENDING)
+    const act = n => getBlocResults(results.map(r => r.hash), chainId, false, n)
```
Comment (dustinnorwood) [inline lib/rest_5.js]: Should be `false, chainId, n`. The old version is wrong too

File: lib/rest_6.js [hunk 2]
```diff
@@ -519,21 +520,14 @@ function* resolveResult(result, options) {
...
-        res = yield getBlocResults(res.map(r => {return r.hash}), options);
-        yield promiseTimeout(1000);
-        count++;
-    }
-
-    if(count >= 60) {
-        throw new HttpError400('Transaction did not resolve');
+function* resolveResults(results) {
+    const pred = res => !res.some(r => r.status === constants.PENDING)
+    const act = node => getBlocResults(results.map(r => r.hash), false, node)
```
Comment (dustinnorwood) [inline lib/rest_6.js]: Needs `options`

File: lib/rest_5.js (hunk 1 above)
Comment (th0114nd) [reply lib/rest_5.js]: Done

File: lib/rest_6.js (hunk 2 above)
Comment (th0114nd) [reply lib/rest_6.js]: Done

**Pull request closed without merging**